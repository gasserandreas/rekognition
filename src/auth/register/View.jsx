import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import {
  AuthHeader,
  AuthFooter,
} from '../AuthComponents';

import CheckEmailForm from './CheckEmailForm';
import RegisterForm from './RegisterForm';

import Card from '../../ui/Card';

import * as Paths from '../../paths';
import { HOCRequestPropTypes } from '../../util/PropTypes';
import { Colors } from '../../styles';


const StyledView = styled.div`
  background-color: ${Colors.ColorsPalette.Background};
  min-height: 100vh;

  .content {
    width: 90%;
    max-width: 400px;
    margin: 0 auto;
    padding: 3rem 0;
  }
`;

class View extends Component {
  static propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    signupRequest: HOCRequestPropTypes.isRequired,
    validEmail: PropTypes.bool,
    signupUser: PropTypes.func.isRequired,
    checkEmail: PropTypes.func.isRequired,
    invalidateEmail: PropTypes.func.isRequired,
  }

  static defaultProps = {
    validEmail: null,
  };

  onCheckEmail = this.onCheckEmail.bind(this);
  onCancelSignUp = this.onCancelSignUp.bind(this);

  state = {
    loading: false,
    email: '',
  }


  componentWillMount() {
    const { isAuthenticated } = this.props;
    this.checkAuth(isAuthenticated);
  }

  componentWillReceiveProps(newProps) {
    const { isAuthenticated } = newProps;

    if (this.props.isAuthenticated !== isAuthenticated) {
      this.checkAuth(isAuthenticated);
    }
  }

  checkAuth(isAuthenticated) {
    if (isAuthenticated) {
      //
      setTimeout(() => {
        this.props.history.push(Paths.HOME);
      }, 300);
    }
  }

  onCheckEmail({ email }) {
    this.setState({ email });
    this.props.checkEmail(email);
  }

  onCancelSignUp() {
    // cancel valid email
    this.props.invalidateEmail();
  }

  render() {
    const {
      signupRequest,
      checkEmailRequest,
      validEmail,
      signupUser,
    } = this.props;

    const { email } = this.state;

    return (
      <StyledView>
        <AuthHeader>Sinup for Rekognition</AuthHeader>
        <div className="content">
          <Card pad="large">
            { !validEmail ? (
              <CheckEmailForm
                user={{ email: '' }}
                onSubmit={this.onCheckEmail}
                submitting={checkEmailRequest.loading}
                error={checkEmailRequest.error ? checkEmailRequest.error.message : null}
                validEmail={validEmail}
              />
            ) : (
              <RegisterForm
                user={{ email, password: '', firstname: '', lastname: '', remember: false }}
                onSubmit={signupUser}
                onCancel={this.onCancelSignUp}
                submitting={signupRequest.loading}
                error={signupRequest.error ? signupRequest.error.message : null}
              />
            )}
          </Card>
        </div>
        <AuthFooter href={Paths.LOGIN}>Use login credentials instead</AuthFooter>
      </StyledView>
    );
  }
}

export default View;
