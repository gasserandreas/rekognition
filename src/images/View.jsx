import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Box } from 'grommet';

import ImageList from './ImageList';
import DetailView from './DetailView';

const StyledImageView = styled(Box)`
  overflow-x: scroll;
`;

class ImageView extends Component {
  static propTypes = {
    images: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  }

  state = {
    expanded: true,
  }

  render() {
    const { images } = this.props;
    const { expanded } = this.state;
    return (
      <StyledImageView
        fill
      >
        <ImageList
          expanded={expanded}
          images={images}
          onImageClick={image => this.setState({ expanded: !expanded })}
        />
        <DetailView
          visible={expanded}
          onClickClose={() => this.setState({ expanded: !expanded })}
        />
      </StyledImageView>
    )
  }
}

export default ImageView;
