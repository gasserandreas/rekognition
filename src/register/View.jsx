/** @jsx jsx */
import { Component } from 'react';
import PropTypes from 'prop-types';
import { jsx, css } from '@emotion/core';
import { Link } from 'react-router-dom';

import CheckEmailForm from './CheckEmailForm';
import RegisterForm from './RegisterForm';

import Card from '../ui/Card';

import * as Paths from '../paths';
import { HOCRequestPropTypes } from '../util/PropTypes';
import { Colors } from '../styles';

const Styles = {
  View: css`
    background-color: ${Colors.Blue.Background};
    color: ${Colors.White.default};
    min-height: 100vh;
  `,
  Content: css`
    width: 90%;
    max-width: 400px;
    margin: 0 auto;
    padding: 3rem 0;
  `,
  PageHeader: css`
    text-transform: uppercase;
    font-size: 2.2rem;
    color: ${Colors.White.default};
    text-align: center;
    margin-bottom: 2rem;
  `,
  LoginLink: css`
    color: ${Colors.White.default};
    text-align: center;
    margin: 2.5rem 0 0;
    display: block;

    &:hover {
      color: ${Colors.White.default};
      text-decoration: underline;
      cursor: pointer;
    }
  `,
}

class View extends Component {
  static propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    signupRequest: HOCRequestPropTypes.isRequired,
    signupUser: PropTypes.func.isRequired,
    checkEmail: PropTypes.func.isRequired,
  }

  onCheckEmail = this.onCheckEmail.bind(this);
  onSignupUser = this.onSignupUser.bind(this);

  state = {
    submitting: false,
    email: '',
  };

  componentWillMount() {
    const { isAuthenticated } = this.props;
    this.checkAuth(isAuthenticated);
  }

  componentWillReceiveProps(newProps) {
    const { isAuthenticated } = newProps;

    if (this.props.isAuthenticated !== isAuthenticated) {
      this.checkAuth(isAuthenticated);
    }
  }

  checkAuth(isAuthenticated) {
    if (isAuthenticated) {
      //
      setTimeout(() => {
        this.props.history.push(Paths.HOME);
      }, 500);
    }
  }

  onCheckEmail(email) {
    this.setState({ email });
    this.props.checkEmail(email);
  }

  onSignupUser({ email_disabled, ...data }) {
    const newData = {
      ...data,
      email: this.state.email,
    };
    
    this.props.signupUser(newData);
  }

  render() {
    const {
      signupRequest,
      checkEmailRequest,
      validEmail,
    } = this.props;
    return (
      <div css={Styles.View}>
        <div css={Styles.Content}>
          <h1 css={Styles.PageHeader}>AWS Rekognition</h1>
          <Card>
            {!validEmail ? (
              <CheckEmailForm
                onSubmit={this.onCheckEmail}
                submitting={checkEmailRequest.loading}
                validEmail={validEmail}
              />
            ) : (
              <RegisterForm
                onSubmit={this.onSignupUser}
                submitting={signupRequest.loading}
                email={this.state.email}
              />
            )}
          </Card>
          <Link
            css={Styles.LoginLink}
            to={Paths.LOGIN}
          >Use login credentials instead</Link>
        </div>
      </div>
    );
  }
}

export default View;
