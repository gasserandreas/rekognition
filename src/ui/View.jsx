import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Box, Heading } from 'grommet';

import { Sizes } from '../styles';

// view
const StyledView = styled(Box)`
  position: relative;
  padding-top: ${Sizes.Header.height};
`;

export const View = ({ children, ...props }) => (
  <StyledView flex fill pad="small" {...props}>
    {children}
  </StyledView>
);

View.propTypes = {
  children: PropTypes.node.isRequired,
};

// heading
const StyledHeading = styled(Heading)`
  margin: 0;
  max-width: 100%;
`;

export const ViewHeading = (props) => (
  <StyledHeading level="4" {...props} />
);

View.propTypes = {
  children: PropTypes.node.isRequired,
};

export default View;
