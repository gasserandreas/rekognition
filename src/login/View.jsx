/** @jsx jsx */
import { Component } from 'react';
import PropTypes from 'prop-types';
import { jsx, css } from '@emotion/core';
import { Link } from 'react-router-dom';

import LoginForm from './LoginForm';

import Card from '../ui/Card';

import * as Paths from '../paths';
import { HOCRequestPropTypes } from '../ui/PropTypes';
import { Colors } from '../styles';

const Styles = {
  View: css`
    background-color: ${Colors.Blue.Background};
    min-height: 100vh;
  `,
  Content: css`
    width: 90%;
    max-width: 400px;
    margin: 0 auto;
    padding: 3rem 0;
  `,
  PageHeader: css`
    text-transform: uppercase;
    font-size: 2.2rem;
    color: ${Colors.White.default};
    text-align: center;
  `,
  SubHeader: css`
    color: ${Colors.White.default};
    margin: 3rem 0;
    text-align: center;
  `,
  SignUpLink: css`
    color: ${Colors.White.default};
    text-align: center;
    margin: 2.5rem 0 0;
    display: block;

    &:hover {
      color: ${Colors.White.default};
      text-decoration: underline;
      cursor: pointer;
    }
  `,
}

class View extends Component {
  static propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    loginRequest: HOCRequestPropTypes.isRequired,
  }

  state = {
    submitting: false,
  };

  componentWillMount() {
    const { isAuthenticated } = this.props;
    this.checkAuth(isAuthenticated);
  }

  componentWillReceiveProps(newProps) {
    const { isAuthenticated } = newProps;

    if (this.props.isAuthenticated !== isAuthenticated) {
      this.checkAuth(isAuthenticated);
    }
  }

  checkAuth(isAuthenticated) {
    if (isAuthenticated) {
      //
      setTimeout(() => {
        this.props.history.push(Paths.HOME);
      }, 500);
    }
  }

  render() {
    const { loginRequest: { error, loading } } = this.props;
    return (
      <div css={Styles.View}>
        <div css={Styles.Content}>
          <h1 css={Styles.PageHeader}>AWS Rekognition</h1>
          <h2 css={Styles.SubHeader}>Login to get started</h2>
          <Card>
            <LoginForm
              onSubmit={this.props.logInUser}
              submitting={loading}
              error={error}
            />
          </Card>
          <Link
            css={Styles.SignUpLink}
            to={Paths.REGISTER}
          >Sign up for an account</Link>
        </div>
      </div>
    );
  }
}

export default View;
