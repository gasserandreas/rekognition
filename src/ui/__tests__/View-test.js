/* global React, shallow, mount, render, renderer, testSnapshot */

import { View, ViewHeading } from '../View';

const getShallowView = (props = { children: 'content' }) => shallow(<View {...props} />);

it('View should render correctly', () => {
  testSnapshot(<View>View Content</View>);
});

it('View should render with content', () => {
  testSnapshot(<View>
    <h1>Hello</h1>
    <p>Some more content, yeaah.</p>
  </View>);
});

it('view should pass additional props to component', () => {
  const props = {
    'data-attr-1': 1,
    'data-attr-2': 2,
    children: 'hello',
  };
  const output = getShallowView(props);

  const compProps = output.props();

  // check for props in compProps
  Object.keys(props)
    .map((key) => {
      const value = props[key];
      expect(compProps[key]).toEqual(value);
    });
});

it('ViewHeading should render correctly', () => {
  testSnapshot(<ViewHeading />);
});

it('ViewHeading should render with content', () => {
  testSnapshot(<ViewHeading>Heading</ViewHeading>);
});
