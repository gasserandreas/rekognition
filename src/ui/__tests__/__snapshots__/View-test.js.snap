// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View should render correctly 1`] = `
<div
  className="sc-bdVaJa dmtOWd StyledBox-sc-13pk1d4-0 cOQfYo"
>
  View Content
</div>
`;

exports[`View should render with content 1`] = `
<div
  className="sc-bdVaJa dmtOWd StyledBox-sc-13pk1d4-0 cOQfYo"
>
  <h1>
    Hello
  </h1>
  <p>
    Some more content, yeaah.
  </p>
</div>
`;

exports[`ViewHeading should render correctly 1`] = `
<h4
  className="sc-bwzfXH cWIyyV StyledHeading-sc-1rdh4aw-0 hlZfma"
/>
`;

exports[`ViewHeading should render with content 1`] = `
<h4
  className="sc-bwzfXH cWIyyV StyledHeading-sc-1rdh4aw-0 hlZfma"
>
  Heading
</h4>
`;
